LXStudio-IDE/src/main/java/titanicsend/app/TEVisual.java

```diff
@@ -0,0 +1,84 @@
+package titanicsend.app;
+
+import java.util.*;
+
+import heronarts.lx.parameter.BooleanParameter;
+import heronarts.p4lx.ui.UI;
+import heronarts.p4lx.ui.UI3dComponent;
+import heronarts.lx.model.LXPoint;
+import processing.core.PGraphics;
+import titanicsend.model.*;
+
+public class TEVisual extends TEUIComponent {
```

Comment:
this class might be a little too generally named -- consider something like TEPreviewPanel?

Reply:
Or TEOverlay or something. Will think it over...

Reply:
yeah that makes sense to me

Reply:
👍 I think "overlay" is a good descriptor. It would also be nice to move the vertex and panel overlays to separate child components at some point.